docs/current_docs/manuals/developer/python/820256-module-structure.mdx

```diff
@@ -5,36 +5,81 @@ toc_max_heading_level: 2
 title: "Module Structure"
 ---
 
-
 # Module Structure
 
-
 It's essential to understand a few key concepts about Dagger Modules created for use with the Python SDK, for a better fit with your normal development workflow.
 
-## Runtime container with Python 3.11
+## Python version
+
+Dagger Modules run in a [container](../overview/942201-execution-environment.mdx) that's bootstrapped specifically by the Python SDK, with the necessary environment to run the module's code. It currently runs Python 3.11 by default but starting with [v0.11.0][v11] that can be changed.
+
+The desired interpreter version can be pinned by creating a `.python-version` file next to your `pyproject.toml`:
+```console
+echo "3.11" > .python-version
+```
+
+This can be useful either to pin the version in case the default one gets an upgrade, or to choose another a newer version sooner.
+
+As an alternative, setting [requires-python](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#python-requires) in `pyproject.toml` will also pin the version, but using a lower bound constraint (minimum version). For example, the following will use 3.11, even though there's a 3.12 available:
+
+```toml
+[project]
+requires-python = ">=3.11"
+```
+
+## Base image
 
-Dagger Modules run in a container (called the "runtime container") that's bootstrapped specifically by the Python SDK, with the necessary environment to run the module's code. It's currently hardcoded to run in Python 3.11 (although this may be configurable in future).
+The default base image is [python:VERSION-slim](https://hub.docker.com/_/python/tags?name=3.11-slim) (based on the [Python version](#python-version)), but it can be overridden in `pyproject.toml` since [v0.11][v11]:
+
+```toml
+[dagger.tool]
+base-image = "acme/python:3.11"
+```
+
+This can be useful to add a few requirements to the module's execution environment, like operating system packages such as `git` for example.
+
+:::warning
+Don't deviate from the default base image too much or it may break in a future version of Dagger. Override at own risk.
+:::
 
 ## Automatic SDK install
 
-The Python SDK is installed automatically, including dependencies, with a version that's tied to the currently running Dagger Engine container[^1]:
+Each release of Dagger bundles the Python SDK inside the Engine container image. During the Dagger Module's code generation phase, that directory is unpacked and installed automatically, including dependencies. That's the copy that is exported to the host on `dagger init --sdk=python` or `dagger develop --sdk=python`. It's **not** the same as installing from [dagger-io](https://pypi.org/project/dagger-io/).
```

Comment:
```suggestion
Each release of Dagger bundles the Python SDK inside the Engine container image. During the Dagger module's code generation phase, that directory is unpacked and installed automatically, including dependencies. It is exported to the host on `dagger init --sdk=python` or `dagger develop --sdk=python`. Note that this is **not** the same as installing from [dagger-io](https://pypi.org/project/dagger-io/).
```

Reply:
The "It is exported to the host" part doesn't seem quite right. I intended to point out or identify that the `./sdk` directory that people see after init comes from this directory that's bundled inside the engine container, and does not come from what's published in PyPI (dagger-io).

Reply:
Updated, see if this is clearer?